lib/ace/ext/papeeria/latex_behaviour.coffee

```diff
@@ -1,9 +1,26 @@
 define((require, exports, module) ->
   Behaviour = require("ace/mode/behaviour").Behaviour
+  COMMENT_REGEX = /comment/
+  ESCAPE_REGEX = /escape/
+  MATH_REGEX = /math/
+  EQUATION_REGEX = /equation/
+  RPAREN_REGEX = /rparen/
+  correspondingClosing = {
+      '(': ')',
+      '[': ']'
+      '{': '}'
+  }
+
+
+  isCommentToken = (token, column) -> token? and COMMENT_REGEX.test(token.type) and column != 0
 
-  isCommentToken = (token) -> token? and /comment/.test(token.type)
 
-  isEscaped = (line, column) -> line[column - 1] == '\\' and line[column - 2] != '\\'
+  isEscapedInsertion = (token, column) -> (
+      token? and
+      ESCAPE_REGEX.test(token.type) and
+      column - token.start == 1
```

Comment:
Here you can't really do it. Before the first check we're not sure if `token` even exists, and before the second check we're not really sure that `token.start` exists.